resolwe_bio/processes/support_processors/multiqc.py

```diff
@@ -0,0 +1,242 @@
+            )
+
+            config = BooleanField(
+                label='Use configuration file',
+                default=True,
+                description='Use Genialis configuration file for MultiQC report.',
+            )
+
+            cl_config = StringField(
+                label='--cl-config',
+                required=False,
+                description='Enter text with command-line configuration options to override the '
+                            'defaults (e.g. custom_logo_url: https://www.genialis.com).',
+            )
+
+        advanced = GroupField(Advanced, label="Advanced options")
+
+    class Output:
+        """Output fields."""
+
+        report = FileHtmlField(label='MultiQC report')
+        report_data = DirField(label='Report data')
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        samples = []
+        species = []
+        build = []
+        unsupported_data = []
+
+        for d in inputs.data:
+            sample_dir = d.entity_name
+
+            if sample_dir and not os.path.isdir(sample_dir):
+                os.mkdir(sample_dir)
+
+            try:
+                if d.entity_name and d.species and d.build:
+                    samples.append(d.entity_name)
+                    species.append(d.species)
+                    build.append(d.build)
+            except AttributeError:
+                pass
+
+            if d.type.startswith('data:reads:fastq:single'):
+                for fq_report in d.fastqc_archive:
+                    name = os.path.basename(fq_report.path)
+                    create_symlink(fq_report.path, os.path.join(sample_dir, name))
+
+            elif d.type.startswith('data:reads:fastq:paired'):
+                for fq_report in d.fastqc_archive + d.fastqc_archive2:
+                    name = os.path.basename(fq_report.path)
+                    create_symlink(fq_report.path, os.path.join(sample_dir, name))
+
+            elif d.type == 'data:alignment:bam:star:':
+                stats_file = os.path.basename(d.stats.path)
+                assert stats_file.endswith('_stats.txt')
+                bam_name = stats_file[:-10]
+
+                if d.build == 'rRNA':
+                    rrna_report = f'{bam_name}.rRNA.Log.final.out'
+                    create_symlink(d.stats.path, os.path.join(sample_dir, rrna_report))
+                elif d.build == 'globin':
+                    globin_report = f'{bam_name}.globin.Log.final.out'
+                    create_symlink(d.stats.path, os.path.join(sample_dir, globin_report))
+                else:
+                    report = f'{bam_name}.Log.final.out'
+                    create_symlink(d.stats.path, os.path.join(sample_dir, report))
+
+            elif d.type.startswith('data:alignment:bam'):
+                name = os.path.basename(d.stats.path)
+                create_symlink(d.stats.path, os.path.join(sample_dir, name))
+
+            elif d.type == 'data:expression:featurecounts:':
```

Comment:
When checking for type, some strings end with a colon while others do not. Is this by design?

Reply:
It does not end with colon where we use `startswith()`, as the colon is not necessary there.